src/ifcopenshell-python/ifcopenshell/util/unit.py

```diff
@@ -445,56 +451,83 @@ def get_property_unit(
     :return: The IFC unit entity, or nothing if there is no default project
         unit defined.
     """
-    unit = getattr(prop, "Unit", None)
-    if unit:
+    unit = prop.Unit
+    if isinstance(unit, ifcopenshell.entity_instance):
         return unit
-    unit_assignment = get_unit_assignment(ifc_file)
-    if not unit_assignment:
-        return
-    entity = prop.wrapped_data.declaration().as_entity()
+
+    value = None
     measure_class = None
+
+    # DEV-NOTE: Using .is_a() is wrong, as it tells us nothing about super class membership
```

Comment:
NOTE: this was more of a note to myself, as I was initially tempted to change this to
```py
match prop.is_a():
  case "IfcPhysicalSimpleQuantity":
    ...
  case "IfcPropertySingleValue":
    ...
  ...
```
(or if-elseif-...) until I realised this does not work, as `.is_a()` returns the current ifc-type, but says nothing about membership to a "super" ifc-type.